src/components/cvaa-overlay/cvaa-overlay.js

```diff
@@ -97,67 +98,139 @@ class CVAAOverlay extends BaseComponent {
+      fontColor: this.props.player.TextStyle.StandardColors.WHITE
+    });
+
     return (
       <div className={this.state.state === cvaaOverlayState.Main ? 'overlay-screen active' : 'overlay-screen'}>
         <div className='title'>
           Advanced captions settings
         </div>
         <div>
-          <div className='sample' onClick={() => this.changeCaptionsStyle('default')}>Sample</div>
-          <div className='sample black-bg' onClick={() => this.changeCaptionsStyle('black-bg')}>Sample</div>
-          <div className='sample yellow-text' onClick={() => this.changeCaptionsStyle('yellow-text')}>Sample</div>
+          <div className='sample' onClick={() => this.changeCaptionsStyle(captionsStyleDefault)}>Sample</div>
+          <div className='sample black-bg' onClick={() => this.changeCaptionsStyle(captionsStyleBlackBG)}>Sample</div>
+          <div className='sample yellow-text' onClick={() => this.changeCaptionsStyle(captionsStyleYellow)}>Sample</div>
         </div>
         <a className='button-save-cvaa' onClick={() => this.transitionToState(cvaaOverlayState.CustomCaptions)}>Set custom caption</a>
       </div>
     )
   }
 
   /**
+   * change one or more properties in customTextStyle object in the internal state
+   *
+   * @param {Object} styleChanges style changes object
+   * @returns {void}
+   * @memberof CVAAOverlay
+   */
+  changeCustomStyle(styleChanges: Object): void {
+    this.setState({customTextStyle: Object.assign(this.state.customTextStyle, styleChanges)});
+  }
+
+  /**
    * render custom captions state
    *
+   * @param {*} props - component props
    * @returns {React$Element} - custom captions elements
    * @memberof CVAAOverlay
    */
-  renderCustomCaptionsState(): React$Element<any> {
-    var speedOptions = [
-      { value: 1, label: 'Auto (360)', active: true },
-      { value: 2, label: '240' },
-      { value: 3, label: '144' }
-    ];
+  renderCustomCaptionsState(props: any): React$Element<any> {
+
+    const fontFamily = this.props.player.TextStyle.FontFamily;
+    const edgeStyles = this.props.player.TextStyle.EdgeStyles;
+    const standardOpacities = props.player.TextStyle.StandardOpacities;
+    const standardColors = props.player.TextStyle.StandardColors;
+
+    var fontSizeOptions = this.props.player.TextStyle.FontSizes.map(size => ({
+      value: size,
+      label: size,
+      active: this.state.customTextStyle.fontSize === size
+    }));
+
+    var fontColorOptions = Object.keys(standardColors).map(key => ({
+      value: standardColors[key],
+      label: key,
+      active:  this.state.customTextStyle.fontColor == standardColors[key]
+    }));
+
+    var fontFamilyOptions = Object.keys(fontFamily).map(key => ({
+      value: fontFamily[key],
+      label: fontFamily[key],
+      active: this.state.customTextStyle.fontFamily == fontFamily[key]
+    }));
+
+    var fontStyleOptions = Object.keys(edgeStyles).map(key => ({
```

Comment:
The font edge change is not shown on the selection menu, and after selecting it keeps showing `NONE`